diff --git a/Vendored/quickjs/quickjs.c b/Vendored/quickjs/quickjs.c
index 0113885..830841d 100644
--- a/Vendored/quickjs/quickjs.c
+++ b/Vendored/quickjs/quickjs.c
@@ -22,6 +22,10 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
+/*
+ * Modifications by D. Priver are released into the public domain.
+ */
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdarg.h>
@@ -54061,3 +54065,12 @@ JS_get_caller_location(JSContext* ctx, const char** filename, const char** funcn
     }
     return 0;
 }
+
+//
+// NOTE(dpriver): This not being exposed was super
+// annoying.
+//
+JSValue
+JS_ArrayPush(JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv){
+    return js_array_push(ctx, this_val, argc, argv, 0);
+    }
diff --git a/Vendored/quickjs/quickjs.h b/Vendored/quickjs/quickjs.h
index 3bf9388..a5993f5 100644
--- a/Vendored/quickjs/quickjs.h
+++ b/Vendored/quickjs/quickjs.h
@@ -22,6 +22,10 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
+/*
+ * Modifications by D. Priver are released into the public domain.
+ */
 #ifndef QUICKJS_H
 #define QUICKJS_H
 
@@ -1042,6 +1046,7 @@ int JS_SetModuleExportList(JSContext *ctx, JSModuleDef *m,
 #undef js_unlikely
 #undef js_force_inline
 
+//
 // NOTE(dpriver): This was not originally in quickjs
 // I added it so I could implement logging
 //
@@ -1049,6 +1054,11 @@ int JS_SetModuleExportList(JSContext *ctx, JSModuleDef *m,
 int
 JS_get_caller_location(JSContext* ctx, const char** filename, const char** funcname, int* line_num);
 
+//
+// NOTE(dpriver): This not being exposed was super annoying.
+JSValue
+JS_ArrayPush(JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv);
+
 #ifdef __cplusplus
 } /* extern "C" { */
 #endif
