diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..f5ca373
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,5 @@
+.obj
+*.a
+*.dylib
+*.so
+*.dll
diff --git a/Makefile b/Makefile
index 49b1f6f..aa00a20 100644
--- a/Makefile
+++ b/Makefile
@@ -48,7 +48,7 @@ prefix=/usr/local
 # use address sanitizer
 #CONFIG_ASAN=y
 # include the code for BigInt/BigFloat/BigDecimal and math mode
-CONFIG_BIGNUM=y
+# CONFIG_BIGNUM=y
 
 OBJDIR=.obj
 
@@ -59,10 +59,21 @@ ifdef CONFIG_WIN32
     CROSS_PREFIX=x86_64-w64-mingw32-
   endif
   EXE=.exe
+  SO=.dll
+  INSTALL_NAME=
 else
   CROSS_PREFIX=
+ifdef CONFIG_DARWIN
+  SO=.dylib
+  EXE=
+  INSTALL_NAME=-install_name @rpath/$@
+else
+  SO=.so
   EXE=
+  INSTALL_NAME=
+endif
 endif
+
 ifdef CONFIG_CLANG
   HOST_CC=clang
   CC=$(CROSS_PREFIX)clang
@@ -166,9 +177,10 @@ endif
 
 all: $(OBJDIR) $(OBJDIR)/quickjs.check.o $(OBJDIR)/qjs.check.o $(PROGS)
 
-QJS_LIB_OBJS=$(OBJDIR)/quickjs.o $(OBJDIR)/libregexp.o $(OBJDIR)/libunicode.o $(OBJDIR)/cutils.o $(OBJDIR)/quickjs-libc.o
+QJS_LIB_OBJS=$(OBJDIR)/quickjs.o $(OBJDIR)/libregexp.o $(OBJDIR)/libunicode.o $(OBJDIR)/cutils.o
+QJS_LIB_OBJS_LIBC=$(QJS_LIB_OBJS) $(OBJDIR)/quickjs-libc.o
 
-QJS_OBJS=$(OBJDIR)/qjs.o $(OBJDIR)/repl.o $(QJS_LIB_OBJS)
+QJS_OBJS=$(OBJDIR)/qjs.o $(OBJDIR)/repl.o $(QJS_LIB_OBJS_LIBC)
 ifdef CONFIG_BIGNUM
 QJS_LIB_OBJS+=$(OBJDIR)/libbf.o 
 QJS_OBJS+=$(OBJDIR)/qjscalc.o
@@ -190,13 +202,13 @@ qjs$(EXE): $(QJS_OBJS)
 qjs-debug$(EXE): $(patsubst %.o, %.debug.o, $(QJS_OBJS))
 	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
-qjsc$(EXE): $(OBJDIR)/qjsc.o $(QJS_LIB_OBJS)
+qjsc$(EXE): $(OBJDIR)/qjsc.o $(QJS_LIB_OBJS_LIBC)
 	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 ifneq ($(CROSS_PREFIX),)
 
 $(QJSC): $(OBJDIR)/qjsc.host.o \
-    $(patsubst %.o, %.host.o, $(QJS_LIB_OBJS))
+    $(patsubst %.o, %.host.o, $(QJS_LIB_OBJS_LIBC))
 	$(HOST_CC) $(LDFLAGS) -o $@ $^ $(HOST_LIBS)
 
 endif #CROSS_PREFIX
@@ -234,6 +246,10 @@ libquickjs.a: $(patsubst %.o, %.nolto.o, $(QJS_LIB_OBJS))
 	$(AR) rcs $@ $^
 endif # CONFIG_LTO
 
+libquickjs$(SO): $(QJS_LIB_OBJS)
+	$(CC) -shared -o $@ $^ $(INSTALL_NAME)
+
+
 repl.c: $(QJSC) repl.js
 	$(QJSC) -c -o $@ -m repl.js
 
@@ -291,7 +307,7 @@ unicode_gen: $(OBJDIR)/unicode_gen.host.o $(OBJDIR)/cutils.host.o libunicode.c u
 
 clean:
 	rm -f repl.c qjscalc.c out.c
-	rm -f *.a *.o *.d *~ unicode_gen regexp_test $(PROGS)
+	rm -f *$(SO) *.a *.o *.d *~ unicode_gen regexp_test $(PROGS)
 	rm -f hello.c test_fib.c
 	rm -f examples/*.so tests/*.so
 	rm -rf $(OBJDIR)/ *.dSYM/ qjs-debug
@@ -304,6 +320,7 @@ install: all
 	ln -sf qjs "$(DESTDIR)$(prefix)/bin/qjscalc"
 	mkdir -p "$(DESTDIR)$(prefix)/lib/quickjs"
 	install -m644 libquickjs.a "$(DESTDIR)$(prefix)/lib/quickjs"
+	install -m644 libquickjs$(SO) "$(DESTDIR)$(prefix)/lib/quickjs"
 ifdef CONFIG_LTO
 	install -m644 libquickjs.lto.a "$(DESTDIR)$(prefix)/lib/quickjs"
 endif
diff --git a/quickjs.c b/quickjs.c
index 48aeffc..94afa1f 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -54034,3 +54034,30 @@ void JS_AddIntrinsicTypedArrays(JSContext *ctx)
     JS_AddIntrinsicAtomics(ctx);
 #endif
 }
+
+//
+// NOTE(dpriver): This was not originally in quickjs
+// I added it so I could implement logging
+//
+int
+JS_get_caller_location(JSContext* ctx, const char** filename, const char** funcname, int* line_num){
+    JSStackFrame* sf = ctx->rt->current_stack_frame;
+    if(sf == NULL)
+        return -1;
+    sf = sf->prev_frame;
+    if(sf == NULL)
+        return -1;
+    if(funcname)
+        *funcname = get_func_name(ctx, sf->cur_func);
+    JSObject* p = JS_VALUE_GET_OBJ(sf->cur_func);
+    if(js_class_has_bytecode(p->class_id)){
+        JSFunctionBytecode* b = p->u.func.function_bytecode;
+        if(b->has_debug) {
+            if(line_num)
+                *line_num = find_line_num(ctx, b, sf->cur_pc - b->byte_code_buf - 1);
+            if(filename)
+                *filename = JS_AtomToCString(ctx, b->debug.filename);
+        }
+    }
+    return 0;
+}
diff --git a/quickjs.h b/quickjs.h
index d4a5cd3..5c33331 100644
--- a/quickjs.h
+++ b/quickjs.h
@@ -1042,6 +1042,13 @@ int JS_SetModuleExportList(JSContext *ctx, JSModuleDef *m,
 #undef js_unlikely
 #undef js_force_inline
 
+// NOTE(dpriver): This was not originally in quickjs
+// I added it so I could implement logging
+//
+// returns -1 on error, without setting exception.
+int
+JS_get_caller_location(JSContext* ctx, const char** filename, const char** funcname, int* line_num);
+
 #ifdef __cplusplus
 } /* extern "C" { */
 #endif
